libwallet/src/slate.rs

```diff
@@ -57,30 +54,34 @@ pub struct PaymentInfo {
 	pub sender_address: DalekPublicKey,
 	#[serde(with = "dalek_ser::dalek_pubkey_serde")]
 	pub receiver_address: DalekPublicKey,
+	#[serde(default = "default_receiver_signature_none")]
 	#[serde(with = "dalek_ser::option_dalek_sig_serde")]
+	#[serde(skip_serializing_if = "Option::is_none")]
 	pub receiver_signature: Option<DalekSignature>,
 }
 
+fn default_receiver_signature_none() -> Option<DalekSignature> {
+	None
+}
+
 /// Public data for each participant in the slate
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct ParticipantData {
-	/// Id of participant in the transaction. (For now, 0=sender, 1=rec)
-	#[serde(with = "secp_ser::string_or_u64")]
-	pub id: u64,
 	/// Public key corresponding to private blinding factor
 	#[serde(with = "secp_ser::pubkey_serde")]
 	pub public_blind_excess: PublicKey,
 	/// Public key corresponding to private nonce
 	#[serde(with = "secp_ser::pubkey_serde")]
 	pub public_nonce: PublicKey,
 	/// Public partial signature
+	#[serde(default = "default_part_sig_none")]
+	#[serde(skip_serializing_if = "Option::is_none")]
 	#[serde(with = "secp_ser::option_sig_serde")]
 	pub part_sig: Option<Signature>,
-	/// A message for other participants
-	pub message: Option<String>,
-	/// Signature, created with private key corresponding to 'public_blind_excess'
-	#[serde(with = "secp_ser::option_sig_serde")]
```

Comment:
These directives and the Serialize trait on the Slate and included structs aren't necessary (only versioned slates are ever de/serialized), will be cleaning this up in a further PR